rag/llm/key_utils.py

```
@@ -31,4 +34,63 @@ def _normalize_replicate_key(key):
+        payload = key
+    elif isinstance(key, str):
+        try:
+            payload = json.loads(key)
+        except (json.JSONDecodeError, TypeError):
+            logging.warning(
+                "Azure-OpenAI key is not valid JSON; expected a JSON object with 'api_key' (and optionally 'api_version') (see conf/models/azure.json).",
+            )
+            raise ModelException(
+                'Azure-OpenAI requires a JSON key with at least \'api_key\'. Example: {"api_key": "...", "api_version": "2024-02-01"}. See conf/models/azure.json for the model class.',
+                retryable=False,
+            )
```

Ruff (0.16.0)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)